manifests/params.pp

```diff
@@ -35,5 +35,13 @@
   Stdlib::HTTPSUrl $candlepin_url = "https://${candlepin_host}:${candlepin_port}/candlepin",
   String[1] $candlepin_client_keypair_group = 'foreman',
   String[1] $postgresql_evr_package = $katello::globals::postgresql_evr_package,
+  Variant[Boolean, Enum['']] $enable_katello_agent = false,
```

Comment:
This could use a `@param`. I'd also explain that the empty string is due to Hiera. That kind of knowledge tends to be useful for people who haven't played with it and find it odd.